aspnet/2-structured-data/Models/DatastoreBookStore.cs

```diff
@@ -105,24 +90,36 @@ public static Entity ToEntity(this Book book)
         /// <returns>A book.</returns>
         public static Book ToBook(this Entity entity)
         {
-            // TODO: Use reflection so we don't have to modify the code every time we add or drop
-            // a property from Book.
             Book book = new Book();
             book.Id = (long)entity.Key.Path.First().Id;
-            book.Title = entity.Properties.GetValue("Title")?.StringValue;
-            book.Author = entity.Properties.GetValue("Author")?.StringValue;
-            book.PublishedDate = entity.Properties.GetValue("PublishedDate")?.DateTimeValue;
-            book.ImageUrl = entity.Properties.GetValue("ImageUrl")?.StringValue;
-            book.Description = entity.Properties.GetValue("Description")?.StringValue;
-            book.CreatedById = entity.Properties.GetValue("CreatedById")?.StringValue;
+            // P1 Having to call ?.StringValue is annoying.  In C++, the Value type would

```

Comment:
That's already covered with explicit conversions:

```csharp
return new Book
{
    Id = entity.Key.Path.Last().Id,
    Title = (string) entity["Title"],
    Author = (string) entity["Author"],
    PublishedDate = (DateTime) entity["PublishedDate"]
    ...
```

Casting to a nullable type (reference type or nullable value type) will give a null value if the input is null, i.e. if the entity doesn't have that property.